tls/s2n_client_extensions.c

```diff
@@ -289,10 +290,14 @@ int s2n_client_extensions_recv(struct s2n_connection *conn, struct s2n_array *pa
             GUARD(s2n_recv_pq_kem_extension(conn, &extension));
             break;
         case TLS_EXTENSION_SUPPORTED_VERSIONS:
-            GUARD(s2n_extensions_client_supported_versions_recv(conn, &extension));
+            if ( s2n_is_tls13_enabled() ) {
```

Comment:
/nit since we are no longer using macros, s2n style doesn't need spaces around the parenthesis